op_builder/cpu/fused_adam.py

```diff
@@ -12,12 +12,31 @@ class FusedAdamBuilder(CPUOpBuilder):
 
     def __init__(self):
         super().__init__(name=self.NAME)
+        self.dtype = None
+
+    def set_dtype(self, dtype):
+        import torch
+        assert (dtype in [torch.bfloat16, torch.half, torch.float32])
+        self.dtype = dtype
+        return self
 
     def absolute_name(self):
         return f'deepspeed.ops.adam.{self.NAME}_op'
 
     def sources(self):
         return ['csrc/cpu/adam/fused_adam.cpp', 'csrc/adam/cpu_adam_impl.cpp']
 
+    def cxx_args(self):
```

Comment:
Given the identical nature of `cxx_args` across multiple ops, is it possible to refactor to avoid duplication?

Reply:
removed all cxx_args changes. code now define BF16 and FP16 dtype in header.